api/v1/type_bootstrap_config.go

```diff
@@ -42,26 +85,63 @@ type BootstrapConfig struct {
+	// The key to be used by the default service account.
+	// If omitted defaults to an auto generated key.
 	ServiceAccountKey               *string `json:"service-account-key,omitempty" yaml:"service-account-key,omitempty"`
 
 	// Seed configuration for external certificates (node-specific)
+
+	// The certificate to be used for the kube-apiserver.
+	// If omitted defaults to an auto generated certificate.
 	APIServerCert     *string `json:"apiserver-crt,omitempty" yaml:"apiserver-crt,omitempty"`
+	// The key to be used for the kube-apiserver.
+	// If omitted defaults to an auto generated key.
 	APIServerKey      *string `json:"apiserver-key,omitempty" yaml:"apiserver-key,omitempty"`
+	// The certificate to be used for the kubelet.
+	// If omitted defaults to an auto generated certificate.
 	KubeletCert       *string `json:"kubelet-crt,omitempty" yaml:"kubelet-crt,omitempty"`
+	// The key to be used for the kubelet.
+	// If omitted defaults to an auto generated key.
 	KubeletKey        *string `json:"kubelet-key,omitempty" yaml:"kubelet-key,omitempty"`
 	KubeletClientCert *string `json:"kubelet-client-crt,omitempty" yaml:"kubelet-client-crt,omitempty"`
 	KubeletClientKey  *string `json:"kubelet-client-key,omitempty" yaml:"kubelet-client-key,omitempty"`
 
-	// ExtraNodeConfigFiles will be written to /var/snap/k8s/common/args/conf.d
+	// Additional files that are uploaded `/var/snap/k8s/common/args/conf.d/<filename>`
+	// to a node on bootstrap. These files can then be referenced by Kubernetes
+	// service arguments.
+	//
+	// The format is `map[<filename>]<filecontent>`.
 	ExtraNodeConfigFiles map[string]string `json:"extra-node-config-files,omitempty" yaml:"extra-node-config-files,omitempty"`
 
-	// Extra args to add to individual services (set any arg to null to delete)
+	// Extra args to add to individual services (set any arg to null to delete).
+
+	// Additional arguments that are passed to the `kube-apiserver` only for that
+	// specific node. Overwrites default configuration. A parameter that is explicitly
+	// set to `null` is deleted. The format is `map[<--flag-name>]<value>`.
 	ExtraNodeKubeAPIServerArgs         map[string]*string `json:"extra-node-kube-apiserver-args,omitempty" yaml:"extra-node-kube-apiserver-args,omitempty"`
+	// Additional arguments that are passed to the `kube-controller-manager` only for
+	// that specific node. Overwrites default configuration. A parameter that is
+	// explicitly set to `null` is deleted. The format is `map[<--flag-name>]<value>`.
 	ExtraNodeKubeControllerManagerArgs map[string]*string `json:"extra-node-kube-controller-manager-args,omitempty" yaml:"extra-node-kube-controller-manager-args,omitempty"`
+	// Additional arguments that are passed to the `kube-scheduler` only for that
+	// specific node. Overwrites default configuration. A parameter that is explicitly
```

Comment:
nit: "Overwrites default configuration" sound like setting this map would remove all default flags and only set the ones that are specified with the map. 
I would just remove this part as you write later that setting a parameter to null removes it.

Reply:
Makes sense. To be honest I took the content as-is from https://github.com/canonical/k8s-snap/blob/ee017318d958b4c7ed1ab78d3cf4d8b7e94559c5/docs/src/snap/reference/bootstrap-config-reference.md.